tls/s2n_connection.h

```diff
@@ -382,6 +379,10 @@ struct s2n_connection {
     uint32_t server_keying_material_lifetime;
 
     struct s2n_post_handshake post_handshake;
+    /* Both the reader and writer can set key_update_pending.
+     * The writer clears it after a KeyUpdate is sent.
+     */
+    s2n_atomic_flag key_update_pending;
```

Comment:
Is it possible for someone to accidentally set key_update_pending without using the atomic methods? Would the compiler actually catch that?

Reply:
Yes. TSAN will catch a non-atomic update to a variable used by multiple threads, and did in my testing. That's also why Cameron suggested making s2n_atomic_flag a structure instead of just a value. You have to really try to update it without the methods (`key_update_pending.val = blah`).